gourmet/gtk_extras/TextBufferMarkup.py

```diff
@@ -78,26 +80,35 @@ def __init__ (self):
         GObject.GObject.__init__(self)
 
     def set_text (self, txt):
+        if isinstance(txt, bytes):
+            # data loaded from the database are bytes, not str
+            txt = txt.decode()
         Gtk.TextBuffer.set_text(self,"")
         try:
-            self.parsed,self.txt,self.separator = Pango.parse_markup(txt,'\x00')
-        except:
-            print('Problem encountered escaping text: "%s"'%txt)
+            self.parsed, self.attrList, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
+        except Exception as e:  # unescaped text (g-markup-error-quark), eg. contains &amp
+            print(f"Problem encountered escaping text: {txt}: {e}")
             import traceback; traceback.print_exc()
             txt=xml.sax.saxutils.escape(txt)
-            self.parsed,self.txt,self.separator = Pango.parse_markup(txt,'\x00')
-        self.attrIter = self.parsed.get_iterator()
+            self.parsed, self.attrList, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
+
+        self.attrIter = self.attrList.get_iterator()
         self.add_iter_to_buffer()
-        while next(self.attrIter):
+        while self.attrIter.next():
             self.add_iter_to_buffer()
 
     def add_iter_to_buffer (self):
-        range=self.attrIter.range()
-        font,lang,attrs = self.attrIter.get_font()
-        tags = self.get_tags_from_attrs(font,lang,attrs)
-        text = self.txt[range[0]:range[1]]
-        if tags: self.insert_with_tags(self.get_end_iter(),text,*tags)
-        else: self.insert_with_tags(self.get_end_iter(),text)
+        start, end = self.attrIter.range()
+        tags = []
+        ret = self.attrIter.get_font(Pango.FontDescription.new(), None, None)
```

Comment:
unlike what's in the doc, `Pango.AttrIterator.get_font` *requires* these arguments!

Reply:
`Pango.FontDescription()`